src/averageAges.js

```diff
@@ -55,10 +82,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const peopleWithMother = people.filter((child) => (
+    people.some((mother) => mother.name === child.mother)
+      && (onlyWithSon ? child.sex === SEX_MALE : true)
+  ));
+
+  return peopleWithMother.reduce((acc, person) => {
+    const mother = people.find((human) => human.name === person.mother);
+    const ageDiff = person.born - mother.born;
+
+    return acc + ageDiff;
+  }, 0) / peopleWithMother.length;
 }
```

Comment:
Do not forget to use destructuring for objects and break large parts of the code into simpler parts.
```suggestion
  const ageDiff = peopleWithMother.reduce((acc, { born, mother: motherName }) => {
    const motherBirthYear = people.find(({ name }) => name === motherName).born;
    return acc + born - motherBirthYear;
  }, 0);
  return ageDiff / peopleWithMother.length;
}
```